src/main/java/de/dennisguse/opentracks/TrackListActivity.java

```diff
@@ -205,7 +205,7 @@ public void onClick(View v) {
             if (!PreferencesUtils.isRecording(recordingTrackId)) {
                 // Not recording -> Recording
                 updateMenuItems(false, true);
-                Intent newIntent = IntentUtils.newIntent(TrackListActivity.this, TrackDetailActivity.class);
+                Intent newIntent = IntentUtils.newIntent(TrackListActivity.this, TrackDetailActivity.class).putExtra(TrackDetailActivity.EXTRA_RECORDING_REQUEST, true);
```

Comment:
That is somehow implicit and hard to see.
Can we somehow make this easier to read?

Reply:
I've reviewed code before resume option and I've followed the same approach. Before, TrackListActivity used a flag called `startNewRecording` to know if user had been clicked on record button to start a recording, so if user clicked on Start GPS button it has not start recording because `startNewRecording == false`.

Now, TrackDetailActivity uses a flag called `isRecordingRequest` to know if it has to start a recording. So, if an Activity wants to start a new recording must starts TrackDetailActivity passing true in the extra argument EXTRA_RECORDING_REQUEST.

Then, if someone want to use TrackDetailActivity should understand what EXTRA_ data is for. Yes, that become code a harder one.

TrackListActivity and TrackDetailActivity need to "speak" each other. The other option is doing so through Service and the logic become harder.

I can't think of anything better, some clue may help. Anyway, I can give this a spin.

---

Offtopic: Maybe https://github.com/OpenTracksApp/OpenTracks/issues/171 is an option to simplify all this. With TrackListActivity, TrackDetailActivity and TrackRecordActivity could be more simple.

Reply:
I just meant the long line :)
It is so easy to overlook this additional putExtra.

#171: I totally agree.